internal/report/snapshot_reporter.go

```diff
@@ -0,0 +1,214 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package report contains reporters for collecting specific information about the collector-
+package report
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/observiq/observiq-otel-collector/internal/report/snapshot"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// snapShotKind is the kind for the snapshot reporter
+var snapShotKind = "snapshot"
+
+// snapshotConfig specifies what snapshots to collect
+type snapshotConfig struct {
+	// Endpoint is where to send the snapshots
+	Endpoint *endpointConfig `yaml:"endpoint"`
+
+	// Processor is the full ComponentID of the snapshot processor
+	Processor string `yaml:"processor"`
+
+	// PipelineType will be "logs", "metrics", or "traces"
+	PipelineType string `yaml:"pipeline_type"`
+}
+
+// endpointConfig is the configuration of a specific endpoint and full headers to include
+type endpointConfig struct {
+	URL     string      `yaml:"url"`
+	Headers http.Header `yaml:"headers"`
+}
+
+var _ Reporter = (*SnapshotReporter)(nil)
+var _ snapshot.Snapshotter = (*SnapshotReporter)(nil)
+
+// SnapshotReporter tracks and reports snapshots
+type SnapshotReporter struct {
+	client Client
+
+	// minPayloadSize is the minimum number of items to be in a payload
+	minPayloadSize int
```

Comment:
This comment and variable name doesn't seem right; This is more like the loose upper bound on the number of payload items, not really the minimum number to send, seems like we send the payload on reconfigure no matter what.

Reply:
Yeah, I'll fix this it was due to multiple iterations on this code.